src/relcast.erl

```diff
@@ -842,16 +842,35 @@ get_mod_state(DB, OldCF, Module, ModuleState0) ->
         not_found ->
             case rocksdb:get(DB, ?stored_module_state, []) of
                 {ok, SerializedModuleState} ->
-                    rehydrate(Module, SerializedModuleState, ModuleState0);
+                    {SerState, ModState, _} = rehydrate(Module, SerializedModuleState, ModuleState0),
+                    {ok, Txn} = transaction(DB, WriteOpts),
+                    New = Module:serialize(ModState),
+                    KT =
+                        case do_serialize(Module, undefined, New, ?stored_key_prefix, Txn) of
+                            bin ->
+                                bin;
+                            KeyTree ->
+                                _ = maybe_write_key_tree(KeyTree, #state{transaction = Txn}),
+                                ok = rocksdb:transaction_delete(Txn, ?stored_module_state),
+                                KeyTree
+                        end,
+                    rocksdb:transaction_commit(Txn),
+                    {SerState, ModState, KT};
                 not_found ->
-                    ModuleState0
+                    case rocksdb:get(DB, ?stored_key_tree, []) of
+                        {ok, KeyTreeBin} ->
+                            KeyTree = binary_to_term(KeyTreeBin),
+                            do_deserialize(Module, ModuleState0, ?stored_key_prefix, KeyTree, DB);
+                        not_found ->
+                            {undefined, ModuleState0, bin}
+                    end
             end
     end.
 
 rehydrate(Module, SerState, ModuleState0) ->
     OldModuleState = Module:deserialize(SerState),
     {ok, RestoredModuleState} = Module:restore(OldModuleState, ModuleState0),
-    RestoredModuleState.
+    {SerState, RestoredModuleState, bin}.
```

Comment:
What is the third element of the tuple for?

Reply:
the keytree, so we can cache it and rewrite it if it has changed because of new code being loaded.

Reply:
I mean the `bin`, it seems unused?